datafusion/physical-plan/src/joins/cross_join.rs

```diff
@@ -47,7 +47,15 @@ use async_trait::async_trait;
 use futures::{ready, Stream, StreamExt, TryStreamExt};
 
 /// Data of the left side
-type JoinLeftData = (RecordBatch, MemoryReservation);
+#[derive(Debug)]
+struct JoinLeftData {
+    /// Single RecordBatch with all rows from the left side
+    merged_batch: RecordBatch,
+    /// Track memory reservation for merged_batch. Relies on drop
+    /// semantics to release reservation when JoinLeftData is dropped.
+    #[allow(dead_code)]
+    reservation: MemoryReservation,
```

Comment:
Apparently this can be also written without disabling a lint:

```suggestion
    _reservation: MemoryReservation,
```

- `#[allow(dead_code)]` has disadvantage that it can be retained in the code even if the field becomes used
- `_reservation` has disadvantage that it doesn't look pretty

Reply:
Pretty or not, `_reservation` is the language-appropriate idiom.

Reply:
Filed https://github.com/apache/datafusion/issues/13278 to track. Thank you for the suggestion